stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/internal/service/CallService.kt

```diff
@@ -490,13 +490,13 @@ internal open class CallService : Service() {
         }
     }
 
-    private fun playCallSound(@RawRes sound: Int?) {
-        sound?.let {
+    private fun playCallSound(soundUri: Uri?, mediaPlayer: MediaPlayer?) {
+        if (soundUri != null && mediaPlayer != null) {
             try {
-                mediaPlayer?.let {
-                    if (!it.isPlaying) {
-                        setMediaPlayerDataSource(it, sound)
-                        it.start()
+                with(mediaPlayer) {
```

Comment:
This part where we are playing the ringtone. Should we not use the `val reingtone = RingtoneManager.getRingtone(context, uri)` and then play the ringtone after getting audio focus.
I think using `MediaPlayer` here is wrong, lets change this.

We can also do this for our own `Uri`s.

```kotlin
fun requestAudioFocus(context: Context): Boolean {
    audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager

    val audioAttributes = AudioAttributes.Builder()
        .setUsage(AudioAttributes.USAGE_NOTIFICATION_RINGTONE)
        .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
        .build()

    focusRequest = AudioFocusRequest.Builder(AudioManager.AUDIOFOCUS_GAIN_TRANSIENT)
        .setAudioAttributes(audioAttributes)
        .setAcceptsDelayedFocusGain(false)
        .setOnAudioFocusChangeListener { /* Handle focus change if needed */ }
        .build()

    val result = audioManager.requestAudioFocus(focusRequest)
    return result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED
}

// Then
fun playRingtone(context: Context) {
    if (requestAudioFocus(context)) {
        val ringtone = getDefaultRingtone(context)
        ringtone.play()
    } else {
        // Handle failure to gain audio focus
    }
}
```

Reply:
Good find, I'll look into it. What I know:
- Using `RingtoneManager` and it's `repeatSound` method requires a Android version bigger than our `minSDK` so for now I just kept it simple.
- It seems that `RingtoneManager` uses `MediaPlayer` under the hood.
- Maybe `requestAudioFocus` helps with reported issues about not playing sound on some devices.

I'll test the volume controls again and look into it. Then decide if/what to leave for a future PR, like we said.